### PR TITLE
Improve random performance and misc. fixes

### PR DESCRIPTION
**Pull request by jonnhy12333**
- Fix issue with not being able to click into search of the emoji popup
- Fix performance a little bit on the 'onChannelMessageCreated'
- Update random logic for chatty by using the REST api that firebase provides / shallow copy of the data
- Chatty can now choose a random message from anywhere in the channel (not just the last 1,000 messages)
- Chatty no longer responds to direct messages (was gonna take a little bit of work to implement and I just don't care / nobody uses this that I'm aware of). We can add in direct message behavior later if someone cares that much

### CODE REVIEW

File: functions/src/index.ts [hunk 1]
```diff
@@ -21,30 +21,33 @@ export const onChannelMessageCreate = functions.database
...
-        const subscription = await admin.database().ref(subscriptionPath).once('value');
-        const isGeneral = context.params.channelId === 'general';
-        const isSubscribed = isGeneral || subscription.val();
-
-        // Only increase the 'unreadMessageCount' when the user is not active at all
-        if (!user.isActive && isSubscribed) {
-          const newCount = user.unreadMessageCount ? user.unreadMessageCount + 1 : 1;
-          updates[`${userSnapshot.key}/unreadMessageCount`] = newCount;
-        }
+        // If this user has more than 100 unread messages, don't worry about updating numbers, they don't care
```
Comment (greggiannone) [inline functions/src/index.ts]: Have we done user surveys to confirm that they indeed "don't care"?!

File: src/app/components/chat-form/chat-form.component.html [hunk 2]
```diff
@@ -63,3 +58,9 @@
 
   </div>
 </form>
+
+<div class="chat-form__emoticon-popup mat-elevation-z4" *ngIf="showEmojis">
+  <emoji-mart #emojiMart [set]="auth.currentEmojiSet$ | async" [autoFocus]="true" class="mat-elevation-z8"
+    [custom]="customEmoji.customEmoticons" (emojiSelect)="onEmojiSelected($event.emoji)" title="Select an emoji…">
```
Comment (greggiannone) [inline src/app/components/chat-form/chat-form.component.html]: I just realized you used the ellipsis character and not 3 periods, interesting

File: src/bots/services/chattybot/chattybot.service.ts [hunk 3]
```diff
@@ -19,60 +19,56 @@ export class ChattybotBotService extends BotBaseService {
...
+    this.botService.getMood$(this).pipe(first()).subscribe(mood => {
+      if (!mood) {
+        this.setRandomMood(args.channelId);
+      }
+    })
     if (args && args.message) {
       if (args.message.message.match(/\bchatty\b/gi)) {
         this.sendRandomMessage(args);
-      } else if (!args.message.primaryTag) {
+      } else if (args.channelId === 'general') {
```
Comment (greggiannone) [inline src/bots/services/chattybot/chattybot.service.ts]: Do we want this? I thought people LOVED chatty in other channels (we could just remove the broken logic I had)

File: src/app/components/chat-form/chat-form.component.html (hunk 2 above)
Comment (jonnhy12333) [reply src/app/components/chat-form/chat-form.component.html]: i actually didn't know i did that...

File: src/bots/services/chattybot/chattybot.service.ts (hunk 3 above)
Comment (jonnhy12333) [reply src/bots/services/chattybot/chattybot.service.ts]: is that right?! could just remove this if statement!?

Comment (greggiannone) [reply src/bots/services/chattybot/chattybot.service.ts]: If we had polls we could poll the users :troll:

File: functions/src/index.ts (hunk 1 above)
Comment (jonnhy12333) [reply functions/src/index.ts]: I had previously done a pew poll.... it checked out that users didn't care.  I wonder though, maybe this number is too low.... not sure. But 100 'unread' messages is different than having hundreds of messages in channels that they haven't gotten to.  To be 'unread' means that you're not even online when the message comes in

**Pull request merged**